rs/nns/governance/src/governance.rs

```diff
@@ -4522,20 +4521,60 @@ impl Governance {
         );
     }
 
-    fn perform_manage_network_economics(&mut self, new_network_economics: NetworkEconomics) {
-        let Some(original_network_economics) = &self.heap_data.economics else {
-            // This wouldn't happen in production, but if it does, we try to do
-            // the best we can, because doing nothing seems more catastrophic.
+    fn perform_manage_network_economics(
+        &mut self,
+        proposal_id: u64,
+        proposed_network_economics: NetworkEconomics,
+    ) {
+        // This does all the "real work". This may look a little strange, but it
+        // ensures that we do NOT forget to call
+        // self.set_proposal_execution_status (e.g. when we return early). let
+        let mut main = || -> Result<(), GovernanceError> {
+            let default_network_economics = NetworkEconomics::with_default_values();
+
+            let original_network_economics = self
+                .heap_data
+                .economics
+                .as_ref()
+                // It does not seem possible for this to occur in practice, but
+                // if it did, it is probably better to make due, rather than
+                // give up.
+                .unwrap_or_else(|| {
+                    println!(
+                        "{}ERROR: No NetworkEconomics. Using default for the purposes \
+                         of executing a ManageNetworkEconomics proposal.",
+                        LOG_PREFIX
+                    );
+
+                    &default_network_economics
+                });
+
+            // Merge (and validate) proposed_network_economics into original.
+            let new_network_economics =
+                proposed_network_economics.inherit_from(original_network_economics);
```

Comment:
The 

```
inherit_from
validate
```

pattern seems to occur more than once, I'd suggest factoring it out into a method on `NetworkEconomics`

You could call that method `NetworkEconomics::validate_delta` or `NetworkEconomics::validate_changes`